trio/_socket.py

```diff
@@ -697,77 +824,144 @@ async def connect(self, address):
+        _stdlib_socket.socket.recvfrom, _core.wait_readable
+    )
 
     ################################################################
     # recvfrom_into
     ################################################################
 
-    recvfrom_into = _make_simple_sock_method_wrapper(
-        "recvfrom_into", _core.wait_readable
+    if TYPE_CHECKING:
+        # return type of socket.socket.recvfrom_into in typeshed is tuple[bytes, Any]
+        def recvfrom_into(
+            __self, buffer: Buffer, nbytes: int = 0, flags: int = 0
+        ) -> Awaitable[tuple[int, Address]]:
+            ...
+
+    recvfrom_into = _make_simple_sock_method_wrapper(  # noqa: F811
+        _stdlib_socket.socket.recvfrom_into, _core.wait_readable
     )
 
     ################################################################
     # recvmsg
     ################################################################
 
     if hasattr(_stdlib_socket.socket, "recvmsg"):
-        recvmsg = _make_simple_sock_method_wrapper(
-            "recvmsg", _core.wait_readable, maybe_avail=True
+        if TYPE_CHECKING:
+
+            def recvmsg(
+                __self, __bufsize: int, __ancbufsize: int = 0, __flags: int = 0
+            ) -> Awaitable[tuple[bytes, list[tuple[int, int, bytes]], int, Any]]:
+                ...
+
+        recvmsg = _make_simple_sock_method_wrapper(  # noqa: F811
+            _stdlib_socket.socket.recvmsg, _core.wait_readable, maybe_avail=True
         )
 
     ################################################################
     # recvmsg_into
     ################################################################
 
     if hasattr(_stdlib_socket.socket, "recvmsg_into"):
-        recvmsg_into = _make_simple_sock_method_wrapper(
-            "recvmsg_into", _core.wait_readable, maybe_avail=True
+        if TYPE_CHECKING:
+
+            def recvmsg_into(
+                __self,
+                __buffers: Iterable[Buffer],
+                __ancbufsize: int = 0,
+                __flags: int = 0,
+            ) -> Awaitable[tuple[int, list[tuple[int, int, bytes]], int, Any]]:
+                ...
+
+        recvmsg_into = _make_simple_sock_method_wrapper(  # noqa: F811
+            _stdlib_socket.socket.recvmsg_into, _core.wait_readable, maybe_avail=True
         )
 
     ################################################################
     # send
     ################################################################
 
-    send = _make_simple_sock_method_wrapper("send", _core.wait_writable)
+    if TYPE_CHECKING:
+
+        def send(__self, __bytes: Buffer, __flags: int = 0) -> Awaitable[int]:
+            ...
+
+    send = _make_simple_sock_method_wrapper(  # noqa: F811
+        _stdlib_socket.socket.send, _core.wait_writable
+    )
 
     ################################################################
     # sendto
     ################################################################
 
+    @overload
```

Comment:
note to self: add `@overload` to list of markers that `Coverage` should ignore the content of